src/ChildProcessWorker.ts

```diff
@@ -0,0 +1,105 @@
+import { fileURLToPath, URL } from "url";
+import { ChildProcess, fork, ForkOptions } from "child_process";
+import { EventEmitter } from "events";
+
+export class ChildProcessWorker {
```

Comment:
This class mimics `Worker` (with `on("message", ...)` and `postMessage(...)`) except it uses `fork` and node's IPC instead.